### PR TITLE
Add a missing 'category' meta-linter rule

### PR DESCRIPTION
**Pull request by minusworld**

### CODE REVIEW

File: yaml/semgrep/missing-category.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  message: >-
+    This Semgrep rule is missing a valid 'category' field in the 'metadata'.
+    'category' must be one of 'security', 'correctness', 'best-practice',
+    'performance', 'maintainability', or 'portability'.
+  severity: INFO
+  patterns:
+  - pattern-inside: 'rules: [..., $RULE, ...]'
+  - pattern: $RULE
+  - pattern-not: |
+      ...
```
Comment (mschwager) [inline yaml/semgrep/missing-category.yaml]: Do we need this `...`? I know hanging `...` at the beginning or end tend to slow things down.

File: yaml/semgrep/missing-category.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+rules:
+- id: missing-category
```
Comment (mschwager) [inline yaml/semgrep/missing-category.yaml]: Should we name this `metadata-category` to be consistent with other rules here?

**Review by mschwager — APPROVED** [review]
LGTM - I left a few nits for consideration

File: yaml/semgrep/missing-category.yaml (hunk 1 above)
Comment (minusworld) [reply yaml/semgrep/missing-category.yaml]: Good question, I didn't test that! brb

Comment (minusworld) [reply yaml/semgrep/missing-category.yaml]: Not needed, good catch 👍

**Pull request merged**